### ISSUE TITLE
[HELD #41619] [Help Wanted] [$250] Chat – No user name in Concierge welcome message

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#41328]
<article class="markdown-body entry-content comment-body p-0">
      <div class="expandable unchanged js-expandable rich-diff-level-zero">
<p class="unchanged rich-diff-level-one">If you haven’t already, check out our <a href="https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md">contributing guidelines</a> for onboarding and email <a href="mailto:contributors@expensify.com">contributors@expensify.com</a> to request to join our Slack channel!</p>
<hr class="unchanged rich-diff-level-one">
<p class="unchanged rich-diff-level-one"><strong>Version Number:</strong> 1.4.68-0<br>
<strong>Reproducible in staging?:</strong> Y<br>
<strong>Reproducible in production?:</strong> Y<br>
<strong>If this was caught during regression testing, add the test name, ID and link from TestRail:</strong> <a href="https://expensify.testrail.io/index.php?/tests/view/4524515" rel="nofollow">https://expensify.testrail.io/index.php?/tests/view/4524515</a><br>
<strong>Email or phone of affected tester (no customers):</strong> <a href="mailto:ponikarchuks+430424@gmail.com">ponikarchuks+430424@gmail.com</a><br>
<strong>Issue reported by:</strong> Applause - Internal Team</p>
<h2 class="unchanged rich-diff-level-one">Action Performed:</h2>
<ol class="unchanged rich-diff-level-one">
<li class="unchanged">Go to <a href="https://staging.new.expensify.com/" rel="nofollow">https://staging.new.expensify.com/</a>
</li>
<li class="unchanged">Log in as a new user</li>
<li class="unchanged">Enter a name in the onboarding modal and press continue</li>
<li class="unchanged">Concierge chat opens</li>
</ol>
<h2 class="unchanged rich-diff-level-one">Expected Result:</h2>
<p class="unchanged rich-diff-level-one">We should address the user by the firstName if we have one.</p>
<h2 class="unchanged rich-diff-level-one">Actual Result:</h2>
<p class="unchanged rich-diff-level-one">We mention the user by their email, instead of just referring to them by firstName, which they just entered during onboarding.</p>
<h2 class="unchanged rich-diff-level-one">Workaround:</h2>
<p class="unchanged rich-diff-level-one">Unknown</p>
<h2 class="unchanged rich-diff-level-one">Platforms:</h2>
<p class="unchanged rich-diff-level-one">Which of our officially supported platforms is this issue occurring on?</p>
<ul class="contains-task-list unchanged rich-diff-level-one">
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox"><label></label> Android: Native</li>
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox"><label></label> Android: mWeb Chrome</li>
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox"><label></label> iOS: Native</li>
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox"><label></label> iOS: mWeb Safari</li>
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox" checked=""><label></label> MacOS: Chrome / Safari</li>
<li class="task-list-item unchanged">
<input type="checkbox" id="" disabled="" class="task-list-item-checkbox"><label></label> MacOS: Desktop</li>
</ul>
<h2 class="unchanged rich-diff-level-one">Screenshots/Videos</h2>
<p class="unchanged rich-diff-level-one">Add any screenshot/video evidence</p>
<p class="unchanged rich-diff-level-one"><a href="https://github.com/Expensify/App/assets/78819774/c018b0e1-5f7a-40d2-82cc-47e4be70227e">https://github.com/Expensify/App/assets/78819774/c018b0e1-5f7a-40d2-82cc-47e4be70227e</a></p>
</div>
<p class="vicinity rich-diff-level-zero"><a href="https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22" class="rich-diff-level-one">View all open jobs on GitHub</a></p>
<ins><details class="rich-diff-level-zero"><summary class="rich-diff-level-one">Upwork Automation - Do Not Edit</summary> <ul class="rich-diff-level-one"> <li>Upwork Job URL: https://www.upwork.com/jobs/~01681aa730088019c9</li> <li>Upwork Job ID: 1786168351516377088</li> <li>Last Price Increase: 2024-05-02</li>
<li>Automatic offers: </li>
<ul>
<li>GandalfGwaihir | Contributor | 0</li>
</ul></li> </ul> </details></ins>
    </article>

<details><summary>Issue Owner</summary>Current Issue Owner: @danielrvidal</details>

**Comment by lanitochka17** [comment]
@strepanier03  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by strepanier03** [comment]
Raised [here](https://expensify.slack.com/archives/C036QM0SLJK/p1714625671434499).

**Comment by strepanier03** [comment]
This is actually working as expected for a public domain user. We could make a change to use their name but this isn't a bug so making that change would be a feature request.

**Comment by abzokhattab** [comment]
i don't understand the issue, can you please explain what is the actual result and what is the desired outcome? as I see that we already show the user handle (email) in the welcome message.

**Comment by rushatgabhane** [comment]
@lanitochka17 do we have a problem solution for this feature request?

**Comment by rushatgabhane** [comment]
ahhh okay, so the name is compulsory now?
Then we can work with your [proposal](https://github.com/Expensify/App/issues/41328#issuecomment-2088300980) @GandalfGwaihir 

🎀 👀 🎀 we'll have to take care of the spacing if any name is empty.

**Comment by neil-marcellini** [comment]
@rushatgabhane @GandalfGwaihir I started a discussion about the UX of this internally and our latest decision is to remove the mention entirely. It's not possible to mention a user by their display name only, and we don't really need a mention.

So the message should be "Hey Gandalf" or "Hey Gandalf Gwaihir".

**Comment by neil-marcellini** [comment]
Heads up, discussions are ongoing for the UX. I'll try to let you know when we reach a more firm decision.

**Comment by neil-marcellini** [comment]
Sorry, but it got blocked by a desire to update the copy and UX for this flow. It probably won't be ready for a while since @danielrvidal is going to be OOO. He'll let you know when it's ready to be worked on again, but for now it's going to be on HOLD.

**Comment by danielrvidal** [comment]
I'm going to move to weekly and will get to it next week!

**Comment by kevinksullivan** [comment]
Updated the OP based on this [convo](https://expensify.slack.com/archives/C03U7DCU4/p1715711249625469)

TLDR:
- Let's hold on https://github.com/Expensify/App/issues/41619 so the GBR is working properly and we can assign tasks to create a GBR in the parent chat, instead of relying on the @ mention
- If we have firstName, let's just update the copy to use that
- Let's not create an @ mention, since we won't need it any longer

**Comment by rushatgabhane** [comment]
still holding on https://github.com/Expensify/App/issues/41619

**Comment by danielrvidal** [comment]
Thanks for updating @kevinksullivan, looks like we're still holding.

**Comment by rushatgabhane** [comment]
still holding

**Comment by danielrvidal** [comment]
We're still holding, that issue looks to be making solid progress still.

**Comment by danielrvidal** [comment]
The other issue is still being worked on so we're still holding. Let's make that work and then address this.

**Comment by danielrvidal** [comment]
I think this has actually be solved already. We do not use the mention in the welcome message at all, it has been moved to the tasks. For gmail accounts, the tasks will remain to be the full email rather than the first name because that is how mentions work. So I think we're good here. 

Please reopen if you disagree.
